src/librustc_session/lint/builtin.rs

```diff
@@ -531,6 +538,7 @@ declare_lint_pass! {
         PATTERNS_IN_FNS_WITHOUT_BODY,
         MISSING_FRAGMENT_SPECIFIER,
         LATE_BOUND_LIFETIME_ARGUMENTS,
+        TYPE_PARAM_ON_VARIANT_CTOR,
```

Comment:
I'm not sure I would include `CTOR` in the name. Also, this isn't limited to *type* parameters.

I'd call it `VARIANT_GENERICS` or `VARIANT_GENERIC_ARGS`.